packages/shorebird_cli/lib/src/commands/patch/macos_patcher.dart

```diff
@@ -0,0 +1,446 @@
+import 'package:shorebird_cli/src/code_signer.dart';
+import 'package:shorebird_cli/src/commands/patch/patch.dart';
+import 'package:shorebird_cli/src/common_arguments.dart';
+import 'package:shorebird_cli/src/doctor.dart';
+import 'package:shorebird_cli/src/executables/executables.dart';
+import 'package:shorebird_cli/src/extensions/arg_results.dart';
+import 'package:shorebird_cli/src/logging/detail_progress.dart';
+import 'package:shorebird_cli/src/logging/shorebird_logger.dart';
+import 'package:shorebird_cli/src/metadata/metadata.dart';
+import 'package:shorebird_cli/src/patch_diff_checker.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/release_type.dart';
+import 'package:shorebird_cli/src/shorebird_artifacts.dart';
+import 'package:shorebird_cli/src/shorebird_documentation.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_flutter.dart';
+import 'package:shorebird_cli/src/shorebird_validator.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+// TODO(bryanoltman): consolidate this - this was copied from [IosPatcher]
+typedef _LinkResult = ({int exitCode, double? linkPercentage});
+
+/// {@template macos_patcher}
+/// Functions to create and apply patches to a macOS release.
+/// {@endtemplate}
+class MacosPatcher extends Patcher {
+  /// {@macro macos_patcher}
+  MacosPatcher({
+    required super.argParser,
+    required super.argResults,
+    required super.flavor,
+    required super.target,
+  });
+
+  String get _patchClassTableLinkInfoPath =>
+      p.join(buildDirectory.path, 'macos', 'shorebird', 'App.ct.link');
+
+  String get _patchClassTableLinkDebugInfoPath =>
+      p.join(buildDirectory.path, 'macos', 'shorebird', 'App.class_table.json');
+
+  String get _aotOutputPath => p.join(buildDirectory.path, 'out.aot');
+
+  String get _appDillCopyPath => p.join(buildDirectory.path, 'app.dill');
+
+  String get _vmcodeOutputPath => p.join(buildDirectory.path, 'out.vmcode');
+
+  /// The name of the split debug info file when the target is macOS.
+  // FIXME: this is only the arm symbols, x64 symbols are at
+  // app.darwin-x86_64.symbols
+  static const splitDebugInfoFileName = 'app.darwin-arm64.symbols';
+
+  /// The additional gen_snapshot arguments to use when building the patch with
+  /// `--split-debug-info`.
+  static List<String> splitDebugInfoArgs(String? splitDebugInfoPath) {
+    return splitDebugInfoPath != null
+        ? [
+            '--dwarf-stack-traces',
+            '--resolve-dwarf-paths',
+            '''--save-debugging-info=${saveDebuggingInfoPath(splitDebugInfoPath)}''',
+          ]
+        : <String>[];
+  }
+
+  /// The path to save the split debug info file.
+  static String saveDebuggingInfoPath(String directory) {
```

Comment:
Leading with a verb makes me think this method has side-effects, rather than being a getter.  🤷

Reply:
agreed can we rename to `splitDebuggingInfoPath`?